src/MessageReader.js

```diff
@@ -32,13 +32,34 @@ class StandardTypeReader {
   buffer: Buffer;
   offset: number;
   view: DataView;
+  _decoder: ?TextDecoder;
+  _decoderStatus: "NOT_INITIALIZED" | "INITIALIZED" | "NOT_AVAILABLE" = "NOT_INITIALIZED";
 
   constructor(buffer: Buffer) {
     this.buffer = buffer;
     this.offset = 0;
     this.view = new DataView(buffer.buffer, buffer.byteOffset);
   }
 
+  _intializeTextDecoder() {
+    if (typeof global.TextDecoder === "undefined") {
+      this._decoderStatus = "NOT_AVAILABLE";
+      return;
+    }
+
+    try {
+      this._decoder = new global.TextDecoder("ascii");
+      this._decoderStatus = "INITIALIZED";
+    } catch (e) {
+      // Swallow the error if we don't support ascii encoding.
+      if (e.message === 'The "ascii" encoding is not supported') {
+        this._decoderStatus = "NOT_AVAILABLE";
+      } else {
+        throw e;
```

Comment:
Is the idea here that we don't want to paper over our own bugs? I'd be tempted to swallow any errors in case a different browser has a different message. We have a fallback path.

Reply:
Yeah, it might be worth doing that.